### ISSUE TITLE
callproc docs misleading

### ISSUE THREAD

**Issue by aklaver** [sqlalchemy/sqlalchemy#7051]
### Describe the bug

The documentation as shown here:

https://docs.sqlalchemy.org/en/14/core/connections.html#calling-stored-procedures

is misleading.  

The implementation of call_proc in different drivers is not standard. In `psycopg2` for instance callproc only works with UDF(user defined functions) not stored procedures. For stored procedures you need to use cur.execute("CALL some_procedure"). It would help to change the documentation to point out that stored procedures and UDF's are two different things and may need to be handled differently depending on the driver.

### To Reproduce

```python
import psycopg2
con = psycopg2.connect(dsn)
cur = con.cursor()
cur.callproc('some_procedure', [1])
```

### Error

Something to effect that a stored procedure needs to  use CALL not SELECT.

### Versions

- OS: Any
- Python: Any
- SQLAlchemy: Any
- Database: Postgres
- DBAPI: Any

**Comment by jvanasco** [comment]
I can see how this is misleading.  

The SQLAlchemy docs example does look very much like the pyscopg2 docs for a function (see https://www.psycopg.org/docs/usage.html?#server-side-cursors) ; psycopg2 doesn't seem to have any docs on stored procedures. The only thing I found was a link to postgres docs in this paragraph: https://www.psycopg.org/docs/usage.html?#transactions-control 

Perhaps the section is renamed to "Functions and Stored Procedures" and expanded to both?

**Comment by zzzeek** [comment]
i think "callproc" does not even apply to every DBAPI, so i think this section needs to have top verbiage where it indicates "note that all DBAPIs have different practices, plesae consult DBAPI documentation for specifics" and then in each section put some more note boxes, "not all DBAPIs use callproc and overall usage details will vary. the following example is only an illustration of how it might look to use a particualr DBAPI function", something like that

**Comment by aklaver** [comment]
Actually psycopg2 does have documentation on stored procedure wrt to callproc. See here [callproc](https://www.psycopg.org/docs/cursor.html?highlight=callproc#cursor.callproc).

**Comment by zzzeek** [comment]
@aklaver  this doesnt matter to SQLAlchemy, the purpose of this section is to illustrate how to break out of SQLAlchemy's APIs and use the specific DBAPI library directly.   The problem here is that this misled you into thinking we are actually documenting how every DBAPI works in this section when this is the opposite of the intent.

**Comment by zzzeek** [comment]
> A) It is not my misunderstanding, I now what callproc does in psycopg2. The filing of this issue was prompted by this SO [Question](https://stackoverflow.com/questions/69247638/python-sqlalchemy-calling-stored-procedurepostgres-with-parameter#comment122395654_69247638), where I was asked to file my response as an issue here.

OK, so *someone* was misled, sorry if that was offensive to you


> 
> B) The title and summary of the section read "Calling Stored Procedures
> For stored procedures with special syntactical or parameter concerns, DBAPI-level callproc may be used:".
> Nowhere is there an indication that callproc usage varies between drivers. This led the OP in the a fore mentioned SO question to assume facts not in evidence. 

please review my proposal at https://github.com/sqlalchemy/sqlalchemy/issues/7051#issuecomment-923105023 :

> i think this section needs to have top verbiage where it indicates "note that all DBAPIs have different practices, plesae consult DBAPI documentation for specifics" and then in each section put some more note boxes, "not all DBAPIs use callproc and overall usage details will vary. the following example is only an illustration of how it might look to use a particualr DBAPI function", something like that


> 
> C) The purpose of the issue was to see if would be possible to give people a heads up that the code may not produce the result it claims in all situations.

i dont see any disagreement on that

**Comment by aklaver** [comment]
> OK, so _someone_ was misled, sorry if that was offensive to you

Oh, I'm not that easily offended:)

> please review my proposal at [#7051 (comment)](https://github.com/sqlalchemy/sqlalchemy/issues/7051#issuecomment-923105023) :
> 
> > i think this section needs to have top verbiage where it indicates "note that all DBAPIs have different practices, plesae consult DBAPI documentation for specifics" and then in each section put some more note boxes, "not all DBAPIs use callproc and overall usage details will vary. the following example is only an illustration of how it might look to use a particualr DBAPI function", something like that
> 

Looks good to me, especially since in psycopg3 callproc goes away per [No callproc](https://www.psycopg.org/psycopg3/docs/basic/from_pg2.html#callproc-is-gone).

**Comment by sqla-tester** [comment]
**jonathan vanasco** has proposed a fix for this issue in the **master** branch:

**Fixes: #7051** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3095

**Comment by aklaver** [comment]
Looks good, just two spelling errors: 1) "The following examples are hyptothetical ..." 2) "... use a particualr DBAPI function."
